include/envoy/config/grpc_mux.h

```diff
@@ -134,6 +134,16 @@ class GrpcMux {
    * @return bool whether the API is paused.
    */
   virtual bool paused(const std::string& type_url) const PURE;
+
+  /**
+   * Initiates an on-demand xDS resource resolution. Currently supports
+   * resolution of VirtualHosts using their associated domain name(s).
+   * @param type_url type URL corresponding to xDS API, currently
+   * only type.googleapis.com/envoy.api.v2.VirtualHost is supported
+   * @param aliases domains or hostname to resolve
+   */
+  virtual void requestAliasesResolution(const std::string& type_url,
```

Comment:
Why can't aliases be internal to `GrpcMux`? I.e. you ask for a regular resource by name, and `GrpcMux` checks internally whether it's known to it by name or by some alias.

Reply:
GrpcMux interface doesn't have a direct way to create/send a DeltaDiscoveryRequest; these are created in response to changes in Watches via addOrUpdateWatch/removeWatch. Resolving an unkown domain name/host or authority header doesn't fit into this: the resource name used to initiate the resolution (route_config_name/domain_name) is transient at this point -- it may or may not map to an actual vhost name (I don't think it makes sense to watch these in any case). 

I need a way to schedule a DeltaDiscoveryRequest that bypasses creation/modification of watches, but perhaps with a different name, without "Aliases" in it.

Reply:
I think it's fine to take a watch on aliases. Inside of `GrpcMux`, we should be able to take the `DeltaDiscoveryResponse` and call the watchers who are listening on aliases as well as the original name.

Reply:
@dmitri-d did we end up deciding to move this to 1st class watches during our discussion at KubeCon?

Reply:
We did not!

Reply:
I think we started to chat about this and moved onto other topics. I recall you explained something along the lines of watches being established once the virtual host is received over VHDS? Can you explain this further?

My main concern with this interface is it's a bit divergent from the normal pub-sub model of xDS. Generally, when you want something via xDS, you ask to subscribe to it, and provide a way to receive updates via callbacks. This is what the watches are essentially (in C++ form). Here we're doing a fire-and-forget request (and possibly asking for multiple things?).